### PR TITLE
Merging Find a course

### PR DESCRIPTION
**Pull request by usmanzaheerkhan**
Find a course post dev merge

### CODE REVIEW

File: SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/Pages/CourseResultsPage.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                        }
+                    }
+                }
+            }
+            return selected;
+        }        
+        public CourseDetailsPage ClickSelectedCourse(int courseNo)
+        {
+            List<IWebElement> listOfCourses = _pageHelper.FindElements(CourseSearchList);
+            _context.Add("CourseHeader", listOfCourses[courseNo - 1].Text);
```
Comment (ilambat) [inline SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/Pages/CourseResultsPage.cs]: Can change the .text to _pageHelper.GetText(listOfCourses[courseNo - 1]);

File: SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/Features/SearchForInvalidCourse.feature [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿Feature: SearchForInvalidCourse
+	
+
+@Findacourse
+Scenario: Search for Invalid Course
+	Given I have navigated to Find a course page
+	When I have entered 'NoCourse' in the course name
+	And I click the Find a course button
+	And an error message should be returned "We didn't find any results for 'NoCourse' with the active filters you've applied. Try searching again."
```
Comment (ilambat) [inline SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/Features/SearchForInvalidCourse.feature]: is this supposed to be 'Then' step? Also you shouldn't be passing any parameters in the Then step. The error message should be validated on the pages class within the method

File: SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/Pages/CourseResultsPage.cs (hunk 1 above)
Comment (usmanzaheerkhan) [reply SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/Pages/CourseResultsPage.cs]: changed

File: SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/Features/SearchForInvalidCourse.feature (hunk 2 above)
Comment (usmanzaheerkhan) [reply SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/Features/SearchForInvalidCourse.feature]: changed as per discussion

File: SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/StepDefinitions/FACCommonStep.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace SFA.DFC.FindACourse.UITests.Project.Tests.StepDefinitions
+{
+    [Binding]
+    public class FACCommonStep
+    {
+        
+        private readonly FindACourseConfig _config;        
+        private readonly IWebDriver _webDriver;
+    public FACCommonStep(ScenarioContext context)
```
Comment (ilambat) [inline SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/StepDefinitions/FACCommonStep.cs]: Can that be aligned correctly

File: SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/StepDefinitions/FACHomePageSteps.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        }
+        [Given(@"I have searched for a course '(.*)' for provider '(.*)' and location '(.*)'")]
+        public void GivenIHaveSearchedForACourseForProviderAndLocation(string strCourseName, string provider, string location)
+        {
+            fACHomePage.EnterSearchCriteria(strCourseName);
+            fACHomePage.EnterProvider(provider);
+            fACHomePage.EnterLocation(location);
+            courseResultsPage = fACHomePage.ClickFindACourse();
+            courseResultsPage.VerifyResults();
+        }
```
Comment (ilambat) [inline SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/StepDefinitions/FACHomePageSteps.cs]: Can All Step classes be refined to be consistent:

Need to have Regions for each GWT..
region Givens
region Whens
region Thens

Have a line break between each step

File: SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/StepDefinitions/FACHomePageSteps.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            fACHomePage.EnterProvider(strProv);
+            fACHomePage.EnterLocation(strLocation);            
+        }
+
+        [Given(@"I have searched for '(.*)'")]
+        public void GivenIHaveSearchedFor(string strCourseName)
+        {
+            fACHomePage.EnterSearchCriteria(strCourseName);
+            courseResultsPage = fACHomePage.ClickFindACourse();
+            courseResultsPage.VerifyResults();
```
Comment (ilambat) [inline SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/StepDefinitions/FACHomePageSteps.cs]: No need for Verify steps within the Given statements

File: SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/Features/SearchForACourseWithOtherDetails.feature [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿Feature: SearchForACourseWithOtherDetails	
+
+@Findacourse
+Scenario: Search for course and other information
+	Given I have navigated to Find a course page
+	And I have searched for a course 'Maths' for provider 'Leicester College' and location 'Leicester'
+	When I change the provider 'Skills' and location 'Birmingham'
+	And I have clicked the Apply Filter button
+	Then the results for the new provider and location should be displayed
```
Comment (ilambat) [inline SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/Features/SearchForACourseWithOtherDetails.feature]: This test is not properly passing. It's asserting incorrectly on the final Then step for location

File: SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/Features/FindACourseandVisitCourseDetailsPage.feature [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+@Findacourse
+Scenario: Find a valid course and check course details page
+	Given I have navigated to Find a course page
+	When I have entered 'Maths' in the course name
+	And I click the Find a course button
+	Then the results for the course should be listed
+	When I click the course no '3'
+	Then I should be taken to the course details page 
+	And I should be able to validate the links
+	And I should be able to click the links to access the information
```
Comment (ilambat) [inline SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/Features/FindACourseandVisitCourseDetailsPage.feature]: This test is a false positive as it's not asserting the course title correctly due to the incorrect context being used

Comment (usmanzaheerkhan) [reply SFA.DFC.TestAutomation/SFA.DFC.FindACourse.UITests/Project/Tests/Features/FindACourseandVisitCourseDetailsPage.feature]: corrected

**Pull request merged**